oioswift/common/middleware/container_hierarchy.py

```diff
@@ -168,9 +206,13 @@ def key(self, account, container, mode, path=None):
         """returns Redis key"""
         ret = self.PREFIX + account + ":" + container + ":"
         if mode:
-            ret += mode + ":"
-            if path:
-                ret += path
+            if self.redis_keys_format == REDIS_KEYS_FORMAT_V1:
```

Comment:
Since this `redis_keys_format` won't change (will it?) during the object's life, can't we define two methods (v1 and v2), and monkey-patch `key` during object creation? Or use inheritance? This would avoid many string comparisons.

Reply:
The problem with inheritance or monkey-patch is that we'll have a lot of duplicated code.
Function like `_build_object_listing()` have only a small difference.